public/app/features/alerting/state/reducers.ts

```diff
@@ -62,11 +62,14 @@ export const initialAlertDefinitionState: AlertDefinitionState = {
     data: [],
     intervalSeconds: 60,
   },
-  queryOptions: { maxDataPoints: 100, dataSource: { name: '-- Mixed --' }, queries: [] },
   queryRunner: new PanelQueryRunner(dataConfig),
   uiState: { ...store.getObject(ALERT_DEFINITION_UI_STATE_STORAGE_KEY, DEFAULT_ALERT_DEFINITION_UI_STATE) },
   data: [],
   alertDefinitions: [] as AlertDefinition[],
+  /* These are functions as they are mutated later on and redux toolkit will Object.freeze state so
+   * we need to store these using functions instead */
+  getInstances: () => [] as DataFrame[],
+  getQueryOptions: () => ({ maxDataPoints: 100, dataSource: { name: '-- Mixed --' }, queries: [] }),
```

Comment:
What is mutable in the query options?

Reply:
expr in prometheus queries

Reply:
where is that mutated?

Reply:
in the query editor.

Reply:
Can't we change that? must be a mistake (or old way of doing things)

Reply:
Sure, I can look into it.

Reply:
But old angular editors do modify it, so an option would be to clone it before we give it to angular editors. But let's keep like this for now